File: services/claude.py
import os

import anthropic

client = anthropic.Anthropic(api_key=os.getenv("CLAUDE_API_KEY"))

MODEL = os.getenv("CLAUDE_MODEL", "claude-sonnet-4-6")

LANG_NAMES = {"ru": "русский", "de": "немецкий"}

SYSTEM_PROMPT = (
    "Ты профессиональный копирайтер. Пишешь живые, структурированные статьи для блога. "
    "Используешь подзаголовки, списки где уместно, и подбираешь подходящий тон под тему. "
    "В конце статьи добавляешь блок из 5-8 хэштегов на том же языке что и статья."
)


def generate_article(topic: str, length: int, lang: str = "ru") -> str:
    lang_name = LANG_NAMES.get(lang, "русский")
    user_prompt = (
        f"Напиши статью на языке: {lang_name}.\n"
        f"Тема: {topic}.\n"
        f"Объём: примерно {length} символов (без учёта хэштегов).\n"
        f"В конце добавь блок из 5–8 хэштегов через пробел, начинающихся с #."
    )
    # Conservative token cap; ~3 chars per token + headroom for hashtags.
    max_tokens = max(800, int(length / 2.5) + 400)

    response = client.messages.create(
        model=MODEL,
        max_tokens=max_tokens,
        system=SYSTEM_PROMPT,
        messages=[{"role": "user", "content": user_prompt}],
    )
    return response.content[0].text

File: src/lib/services/claude.ts
import Anthropic from '@anthropic-ai/sdk'

const client = new Anthropic({ apiKey: process.env.CLAUDE_API_KEY })

const MODEL = process.env.CLAUDE_MODEL ?? 'claude-sonnet-4-6'

const SYSTEM_PROMPT = `Ты профессиональный копирайтер для компании по электромонтажным работам.
Пиши экспертные, полезные статьи. Структурируй текст с заголовками и абзацами.
В конце добавляй 3-5 релевантных хэштегов.`

export async function generateArticle(
  topic: string,
  length: number,
  lang: 'ru' | 'de' = 'ru',
): Promise<string> {
  const maxTokens = Math.max(800, Math.floor(length / 2.5) + 400)
  const langLabel = lang === 'ru' ? 'русском' : 'немецком'

  const response = await client.messages.create({
    model: MODEL,
    max_tokens: maxTokens,
    system: SYSTEM_PROMPT,
    messages: [
      {
        role: 'user',
        content: `Напиши статью на ${langLabel} языке на тему: "${topic}". Объём: около ${length} символов.`,
      },
    ],
  })

  const block = response.content[0]
  if (block.type !== 'text') throw new Error('Unexpected response type from Claude')
  return block.text
}
